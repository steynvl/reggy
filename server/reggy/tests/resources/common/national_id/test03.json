{
  "type": "National ID",
  "information": {
    "kind": "US social security number"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
