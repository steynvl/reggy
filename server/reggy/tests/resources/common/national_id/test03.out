Java
Pattern regex = Pattern.compile("(?!000|666)[0-8]\\d{2}-(?!00)\\d{2}-(?!0000)\\d{4}");
(?!000|666)[0-8]\\d{2}-(?!00)\\d{2}-(?!0000)\\d{4}
Perl
my $regex = /(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}/;
(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}
POSIX
(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}
(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}
Python
regex = re.compile(r'(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}')
(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}
JavaScript
const regex = /(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}/;
(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}
PHP
$regex = '/(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}/';
(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}
Golang
regex, _ := regexp.Compile("(?!000|666)[0-8]\\d{2}-(?!00)\\d{2}-(?!0000)\\d{4}")
(?!000|666)[0-8]\\d{2}-(?!00)\\d{2}-(?!0000)\\d{4}
Rust
let re = Regex::new(r"(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}").unwrap();
(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}
C#
var re = new Regex(@"(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}");
(?!000|666)[0-8]\d{2}-(?!00)\d{2}-(?!0000)\d{4}
Scala
val re = "(?!000|666)[0-8]\\d{2}-(?!00)\\d{2}-(?!0000)\\d{4}".r
(?!000|666)[0-8]\\d{2}-(?!00)\\d{2}-(?!0000)\\d{4}
Kotlin
val regex = Regex("(?!000|666)[0-8]\\d{2}-(?!00)\\d{2}-(?!0000)\\d{4}")
(?!000|666)[0-8]\\d{2}-(?!00)\\d{2}-(?!0000)\\d{4}